File: PRAKTIKUM PEMROGRAMAN I P1/MODUL 6/SOAL 4/PRAK604-2410817210002-Sebastian_Abe_Santoso.c
#include <stdio.h>
#include <string.h>
int main() {
    char kode[100], pesan[100];
    int stars = 0, hashes = 0;
    fgets(kode, sizeof(kode), stdin);
    fgets(pesan, sizeof(pesan), stdin);
    kode[strcspn(kode, "\n")] = '\0';
    pesan[strcspn(pesan, "\n")] = '\0';
    if (strlen(kode) != strlen(pesan)) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return 0;
    } char result[strlen(kode) + 1];
    result[strlen(kode)] = '\0';
    for (int i = 0; i < strlen(kode); i++) {
        if (kode[i] == ' ' && pesan[i] == ' ') {
            result[i] = ' ';
        } else if (kode[i] == pesan[i]) {
            result[i] = '*';
            stars++;
        } else {
            result[i] = '#';
            hashes++;
        }
    } printf("%s\n", result);
    printf("* = %d\n", stars);
    printf("# = %d\n", hashes);
    if (stars >= hashes) {
        printf("Pesan Asli\n");
    } else {
        printf("Pesan Palsu\n");
    } return 0;
} 

File: MODUL 6/SOAL 4/PRAK604-2410817210002-Sebastian_Abe_Santoso.py
bintang = 0; tagar = 0
kode = input()
pesan = input()
if len(kode) != len(pesan):
    print("Panjang kalimat berbeda, pesan palsu")
    exit()
result = []
for i in range(len(kode)):
    if kode[i] == ' ' and pesan[i] == ' ':
        result.append('')
    elif kode[i] == pesan[i]:
        result.append('*')  
        bintang += 1
    else:
        result.append('#')  
        tagar += 1
print("".join(result))  
print("* =", bintang)
print("# =", tagar)
if bintang >= tagar:
    print("Pesan Asli")
else:
    print("Pesan Palsu")